envoy/access_log/access_log_config.h

```diff
@@ -30,15 +29,29 @@ template <class Context> class ExtensionFilterFactoryBase : public Config::Typed
    * @return an instance of extension filter implementation from a config proto.
    */
   virtual FilterBasePtr<Context>
-  createFilter(const Protobuf::Message& config,
+  createFilter(const envoy::config::accesslog::v3::ExtensionFilter& config,
                Server::Configuration::CommonFactoryContext& context) PURE;
 
   std::string category() const override { return category_; }
 
 private:
+  std::string categoryByType() {
+    if constexpr (std::is_same_v<Context, Formatter::HttpFormatterContext>) {
+      // This is a special case for the HTTP formatter context to ensure backwards compatibility.
+      return "envoy.access_loggers.extension_filters";
+    } else {
```

Comment:
nit: no need for else, since the if already returned. By eliminating the else, we remove a layer of nesting for this code.

Reply:
Note this is a `constexpr if`. If we remove the `else`, the compiler will still try to compile the `fmt::format("envoy.{}.access_loggers.extension_filters", Context::category())` and because the HttpFormatterContext doesn't provide the `category()` method, there would be a compile error.

Reply:
Wow! That's amazing. Ok, fair enough.